src/components/organisms/NewResourceWizard/NewResourceWizard.tsx

```diff
@@ -146,6 +210,14 @@ const NewResourceWizard = () => {
     setSubmitDisabled(!defaultValues?.name && !defaultValues?.kind && !defaultValues?.apiVersion);
   }, [defaultInput, defaultValues, isFolderOpen]);
 
+  useEffect(() => {
+    if (savingDestination !== 'saveToFolder') {
+      return;
+    }
+
+    generateExportFileName();
+  }, [selectedFolder]);
```

Comment:
Here we should also add the `savingDestination` as a dependency, otherwise the first time you switch to `saveToFolder` the name won't be generated